examples/ec2/natgateway.yaml

```diff
@@ -1,14 +1,16 @@
-apiVersion: ec2.aws.crossplane.io/v1alpha1
+apiVersion: ec2.aws.crossplane.io/v1beta1
 kind: NATGateway
 metadata:
   name: sample-natgateway
 spec:
   forProvider:
     region: us-east-1
-    allocationIdRef:
-      name: sample-eip
-    subnetIdRef:
-      name: sample-subnet1
+    allocationId: "eipalloc-08f9f35936f4e4fc2"
+#   allocationIdRef:
+#     name: sample-eip
+    subnetId: "subnet-0b9b78fa81d457876"
+#   subnetIdRef:
+#     name: sample-subnet1
```

Comment:
What's the thinking behind replacing these references? I think @muvaf mentioned a preference for having the references point to resources with corresponding examples, which sounds like a good approach to me.

Reply:
I also agree indirect references make better examples. For full transparency into how I got things working, I opened this PR with the exact changes I made. One exception is the change in `examples/kms/key.yaml`, where you need to grant the proper access rights to the IAM principal in the key policy, which in the context of these tests were my own principal. Instead of my user's IAM, I went with the example "root" user.

So in certain places, using plain names/ids of AWS resources instead of references to MRs bought me some time (I did not have to provision them via managed resources, I could reference already existing ones in the playground project or lazily provisioned from the UI). 

Having said this, I think we could devise a more formal process for running these tests, where the env is standardized:
We provision MRs in dependency order, e.g., in order to test acm-pca or a cloudfront distribution, or a bucket policy, you need a bucket first. For some dependents the constraints on a common bucket could be conflicting (and need different parent in dep. graph) but for some others a common bucket could be leveraged. I think we already have a vague standardization in the examples (like most, if not all, refer to a providerconfig named `example`) but we may consider documenting or even scripting the provisioning of a "conformant" environment (set of parent resources either provisioned as MRs or if we have missing types provisioned through cloud platform tools). 

Having documented these tested versions, I will revert such transformations, and only keep "real" changes.